doc/rst/language/spec/procedures.rst

```diff
@@ -707,6 +707,13 @@ transfer or share ownership if those arguments apply to ``owned`` or
       owned SomeClass
       true
 
+If the default intent or ``const`` intent is used for an
+:type:`~OwnedObject.owned` or :type:`~SharedObject.shared` argument, then the
+compiler is allowed to optimize the procedure assuming that the
+:type:`~OwnedObject.owned` or :type:`~SharedObject.shared` actual is not
+changing. For example, the compiler could insert a ``borrowed`` temporary and
+replace all uses of the formal with it.
```

Comment:
I do not like the emphasis on the compiler in a specification. My suggestion:

, then the actual argument is assumed to remain unchanged during the call. If the actual argument changes during the call, the behavior is undefined.